core/src/executor/alter/table.rs

```diff
@@ -5,184 +5,158 @@ use {
+                        });
+                let column_types = match column_types {
+                    Continue(column_types) => column_types,
+                    Break(column_types) => column_types?,
+                };
+                let column_defs = column_types
+                    .iter()
+                    .map(|column_type| match column_type {
+                        Some(column_type) => column_type.to_owned(),
+                        None => DataType::Text,
+                    })
+                    .enumerate()
+                    .map(|(i, data_type)| ColumnDef {
+                        name: format!("column{}", i + 1),
+                        data_type,
+                        nullable: true,
+                        default: None,
+                        unique: None,
+                    })
+                    .collect::<Vec<_>>();
+
+                Some(column_defs)
+            }
+        },
+        None if !column_defs.is_empty() => Some(column_defs.to_vec()),
+        None => None,
+    };
 
+    if let Some(column_defs) = target_columns_defs.as_deref() {
+        validate_column_names(column_defs)?;
+
+        for column_def in column_defs {
+            validate(column_def)?;
+        }
+    }
+
+    if storage.fetch_schema(target_table_name).await?.is_none() {
         let schema = Schema {
             table_name: target_table_name.to_owned(),
             column_defs: target_columns_defs,
             indexes: vec![],
             created: Utc::now().naive_utc(),
         };
 
-        if let Some(column_defs) = schema.column_defs.as_deref() {
-            validate_column_names(column_defs)?;
-
-            for column_def in column_defs {
-                validate(column_def)?;
-            }
-        }
-
-        match (
-            storage.fetch_schema(&schema.table_name).await?,
-            if_not_exists,
-        ) {
-            (None, _) => Ok(Some(schema)),
-            (Some(_), true) => Ok(None),
-            (Some(_), false) => {
-                Err(AlterError::TableAlreadyExists(schema.table_name.to_owned()).into())
-            }
-        }
-    })()
-    .await;
-
-    let storage = match schema.try_self(storage)? {
-        (storage, Some(schema)) => storage.insert_schema(&schema).await?.0,
-        (storage, None) => storage,
-    };
+        storage.insert_schema(&schema).await?;
+    } else if !if_not_exists {
+        return Err(AlterError::TableAlreadyExists(target_table_name.to_owned()).into());
+    }
 
     match source {
-        Some(q) => {
-            let (storage, rows) = async {
-                select(&storage, q, None)
-                    .await?
-                    .map_ok(Into::into)
-                    .try_collect()
-                    .await
-            }
-            .await
-            .try_self(storage)?;
-
-            storage.append_data(target_table_name, rows).await
+        Some(query) => {
+            let rows = select(storage, query, None)
```

Comment:
The `select` function is getting the dyn reference from GStore. I need to fix it here too.